js/models/asteroid.js

```diff
@@ -5,16 +5,52 @@ function Asteroid(x, y, dx, dy, size) {
     this.dy = dy;
     this.size = size;
     this.points = [];
+    this.points[0] = [];
+    this.points[1] = [];
 
-    this.generateShape = function(numberOfSides) {       
-        for (var i = 0; i <= numberOfSides; i += 1) {
-            this.points[i] = [this.x + (this.size + 20 * (Math.floor((Math.random() * 3) + 1))) * Math.cos(i * 2 * Math.PI / numberOfSides),
-            y + (this.size + 20 * (Math.floor((Math.random() * 3) + 1))) * Math.sin(i * 2 * Math.PI / numberOfSides)];
+    this.generateShape = function(numSides) {       
+        for (var i = 0; i <= numSides; i += 1) {
+            this.points[0][i] = this.x + (this.size * 20) * Math.cos(i * 2 * Math.PI / numSides);
+            this.points[1][i] = y + (this.size * 20) * Math.sin(i * 2 * Math.PI / numSides);
         }
+    }
+
+    this.generateShape(randomNumber(5, 10));
+}
 
-        this.points[numberOfSides] = [this.points[0][0], this.points[0][1]];
 
+// Return array of x and y outside of the window for spawning
+function genOutsideWin(shpSize){
+    var arr = [];
+    var canvasX = canvas.width;
+    var canvasY = canvas.height;
+    // Left or Right from 0
+    arr[0] = Math.round(Math.random()) * 2 - 1;
+    // Up or Down from 0
+    arr[1] = Math.round(Math.random()) * 2 - 1;
+    // If both are negative, set one lower than screen size
+    if(arr[0] < 1 && arr[1] < 1){
+        Math.random() > 0.5 ? arr[0] = canvasX - shpSize : arr[1] = canvasY - shpSize;
```

Comment:
Use an if else statement for this.  Generally, you should only use a ternary to assign a variable one of two values. ex: var isTrue = (1 == 2) ? true : false;